src/partition.rs

```diff
@@ -134,11 +136,29 @@ impl Partition {
 
     /// Return the starting offset (in bytes) of this partition.
     pub fn bytes_start(&self, lb_size: disk::LogicalBlockSize) -> Result<u64> {
-        let len = self.first_lba
+        let len = self
+            .first_lba
             .checked_mul(lb_size.into())
             .ok_or_else(|| Error::new(ErrorKind::Other, "partition start overflow - bytes"))?;
         Ok(len)
     }
+
+    /// Check whether this partition is in use
+    pub fn is_used(&self) -> bool {
+        self.part_type_guid.guid != uuid::Uuid::nil()
+    }
+
+    /// Return the number of sectors in the partition
+    pub fn size(&self) -> Result<u64> {
+        if self.first_lba < self.last_lba {
```

Comment:
This is basically a `checked_sub`, which the compiler can optimize.

Reply:
Sure I'll change it